examples/OneDShallowWaterGeostrophicAdjustment.jl

```diff
@@ -0,0 +1,197 @@
+# ∂v/∂t + u∂v/∂x + fu = - D,            (2)
+# ∂η/∂t + ∂u/∂x = - D.                  (3)
+#
+# where D indicates a hyperviscous operator of
+# the form ν*Laplacian^nν, where ν is the viscosity
+# coefficient and nν is the order of the operator.
+# The code time-steps Equations (1-3) simultaneously.
+#
+# The initial conditions prescribe a Gaussian bump larger
+# than the deformation radius in a fluid at rest, with
+# some random noise smaller than the deformation radius
+# added on top. The system develops geostrophically-balanced
+# jets around the Gaussian bump, while the smaller-scale noise
+# propagates around as inertia-gravity waves.
+
+using FourierFlows, PyPlot
+using FFTW: rfft, irfft, fft, ifft
+using LinearAlgebra: mul!, ldiv!
+using Printf
+using Random
+
+nothingfunction(args...) = nothing
+
+function calcN!(N, sol, t, s, v, p, g)
+  mul!(v.uh, g.rfftplan, deepcopy(v.u))
+  mul!(v.vh, g.rfftplan, deepcopy(v.v))
+  mul!(v.ηh, g.rfftplan, deepcopy(v.η))
+  ldiv!(v.ux, g.rfftplan, -im*g.kr.*v.uh)
+  ldiv!(v.vx, g.rfftplan, -im*g.kr.*v.vh)
+  mul!(v.uuxh, g.rfftplan, v.u.*v.ux)
+  mul!(v.uvxh, g.rfftplan, v.u.*v.vx)
+  rhsu = p.f*v.vh + p.g*p.h*im*g.kr.*v.ηh - v.uuxh
+  rhsv = - p.f*v.uh - v.uvxh
+  rhsη = im*g.kr.*v.uh
+  N[:, 1] .= rhsu
+  N[:, 2] .= rhsv
+  N[:, 3] .= rhsη
+  dealias!(N, g, g.kralias)
+  nothing
+end
+
+struct Vars{Aphys, Atrans} <: AbstractVars
+        u :: Aphys
+        v :: Aphys
+        η :: Aphys
+       ux :: Aphys
+       vx :: Aphys
+       uh :: Atrans
+       vh :: Atrans
+       ηh :: Atrans
+     uuxh :: Atrans
+     uvxh :: Atrans
+end
+
+function Vars(::Dev, g::AbstractGrid{T}) where {Dev, T}
+  @devzeros Dev T g.nx u v η ux vx
+  @devzeros Dev Complex{T} (g.nkr) uh vh ηh uuxh uvxh
+  mul!(uh, g.rfftplan, u)
+  mul!(vh, g.rfftplan, v)
+  mul!(ηh, g.rfftplan, η)
+  Vars(u, v, η, ux, vx, uh, vh, ηh, uuxh, uvxh)
+end
+
+struct Params{T} <: AbstractParams
+       ν :: T         # Hyperviscosity coefficient
+      nν :: Int       # Order of the hyperviscous operator
+      g  :: T         # Gravitational acceleration
+      h  :: T         # Fluid depth
+      f  :: T         # Coriolis parameter
+end
+Params(ν, nν, g, h, f) = Params(ν, nν, g, h, f)
+
+function Equation(p, g)
+  LC = zeros(Float64, g.nkr, 3)
+  D = - p.ν*g.kr.^(2*p.nν)
+  LC[:, 1] .= D # u
+  LC[:, 2] .= D # v
+  LC[:, 3] .= D # η
+  FourierFlows.Equation(LC, calcN!, g)
+end
+
+"""
+    updatevars!(prob)
+Update the vars in v on the grid g with the solution in sol.
+"""
+function updatevars!(prob)
+  v, g, sol, p = prob.vars, prob.grid, prob.sol, prob.params
+  ldiv!(v.u, g.rfftplan, deepcopy(sol[:, 1]))
+  ldiv!(v.v, g.rfftplan, deepcopy(sol[:, 2]))
+  ldiv!(v.η, g.rfftplan, deepcopy(sol[:, 3]))
+  nothing
+end
+
+function set_uvη!(u0, v0, η0, prob)
+    g, sol, p, v = prob.grid, prob.sol, prob.params, prob.vars
+    mul!(v.uh, g.rfftplan, u0)
+    mul!(v.vh, g.rfftplan, v0)
+    mul!(v.ηh, g.rfftplan, η0)
+    sol[:, 1] = rfft(u0)
```

Comment:
```suggestion
    sol[:, 1] .= v.uh
    sol[:, 2] .= v.vh
    sol[:, 3] .= v.ηh
```

Since you computed the Fourier transforms of `u0`, `v0`, and `η0` in the lines above you don't need to use `rfft`.